### ISSUE TITLE
Discovery:  Use WASM backend

### ISSUE THREAD

**Issue by jzombie** [getify/youperiod.app#60]
I just happened to stumble upon this video:  [Building a Containerless Future with WebAssembly](https://www.youtube.com/watch?v=vqBtoPJoQOE).

It might be related to what you're building here.

WebAssembly is the future of distributed computing. Its security, memory isolation, small footprint, and true portability are all advantages on the web, but become truly game-changing when used to build functions and services deployed in the cloud. This session illustrates how to host WebAssembly modules in Rust code, how to build modules in many different languages (including pros and cons of each), and how to securely grant cloud-native capabilities to these modules. Discussed in detail is the current state of the art in WebAssembly and what can be built with it today. Learn what developers can start doing now to build the containerless future where WebAssembly modules are the de-facto unit of immutable deployment in the cloud, at the edge, and even in IoT and embedded devices.

**Comment by getify** [comment]
I don't think this relates to what we're up to. For one, we're not going to have a backend at all. For another, we want the frontend client to be as open and code-visible as possible, so sticking most or all of our code into WASM seems counter-productive.

We are already using an inline WASM module for the password hashing (Argon2), but I don't see us doing much more with WASM in this project, at least not from what I can currently see/plan.

**Comment by getify** [comment]
If I'm missing something, I'm happy to be corrected. We already do the password hashing (argon2) in WASM, and we do the encryption using the built-in webcrypto. And we have all the appropriate security headers applied from the server to prevent any sort of XSS, CORS, unauthorized code (CSP and Subresource Integrity).

Can you elaborate on what you imagine could be more secure about our current client code if we used more WASM?

**Comment by jzombie** [comment]
Also, I think you should be encrypting all of the data going over postMessage as well, if you're not doing so, and checking the origin on the receiving side.

To really trust a system, it would have to be majorly locked down in my opinion.

I say this because I have such a need for one myself.
